Revit_Core_Engine/Convert/Revit/FromRevit/ViewFilter.cs

```diff
@@ -0,0 +1,523 @@
+                //Convert the Revit Filter Rule objects into corresponding BHoM FilterRule objects and assign them to the BHoM ViewFilter objects
+                viewFilter.Rules = FilterRulesFromRevit(revitViewFilter, revitFilterRules);
+            }
+
+            //Set identifiers, parameters & custom data
+            viewFilter.SetIdentifiers(revitViewFilter);
+            viewFilter.CopyParameters(revitViewFilter, settings.MappingSettings);
+            viewFilter.SetProperties(revitViewFilter, settings.MappingSettings);
+            refObjects.AddOrReplace(revitViewFilter.Id, viewFilter);
+            return viewFilter;
+        }
+
+        /***************************************************/
+
+        //Convert the Revit Filter Rule objects into corresponding BHoM FilterRule objects and store them in a List - via STREAMS
+        private static List<oM.Revit.FilterRules.FilterRule> FilterRulesFromRevit(this ParameterFilterElement revitViewFilter, List<Autodesk.Revit.DB.FilterRule> revitFilterRules)
+        {
+            List<oM.Revit.FilterRules.FilterRule> bhomFilterRules = revitFilterRules.Select(revitRule =>
+            {
+                TextComparisonType bhomTextEvaluator = 0;
+                NumberComparisonType bhomNumericEvaluator = 0;
+
+                // FILTER STRING RULE
+                if (revitRule.GetType()== typeof(Autodesk.Revit.DB.FilterStringRule))
+                { return (oM.Revit.FilterRules.FilterRule)FilterStringRuleFromRevit(revitViewFilter, revitRule, bhomTextEvaluator); }
+                // FILTER DOUBLE RULE
+                else if (revitRule.GetType()==typeof(Autodesk.Revit.DB.FilterDoubleRule))
+                { return (oM.Revit.FilterRules.FilterRule)FilterDoubleRuleFromRevit(revitViewFilter, revitRule, bhomNumericEvaluator); }
+                // FILTER INTEGER RULE
+                else if (revitRule.GetType() == typeof(Autodesk.Revit.DB.FilterIntegerRule))
+                { return (oM.Revit.FilterRules.FilterRule)FilterIntegerRuleFromRevit(revitViewFilter, revitRule, bhomNumericEvaluator); }
+                // FILTER ELEMENTID RULE
+                else if (revitRule.GetType() == typeof(Autodesk.Revit.DB.FilterElementIdRule))
+                { return (oM.Revit.FilterRules.FilterRule)FilterElementIdRuleFromRevit(revitViewFilter, revitRule, bhomNumericEvaluator); }
+                // FILTER CATEGORY RULE
+                else if (revitRule.GetType() == typeof(Autodesk.Revit.DB.FilterCategoryRule))
+                { return (oM.Revit.FilterRules.FilterRule)FilterCategoryRuleFromRevit(revitViewFilter, revitRule); }
+                // FILTER PARAMETER VALUE PRESENCE RULE
+                else if (revitRule.GetType() == typeof(Autodesk.Revit.DB.ParameterValuePresenceRule))
+                { return (oM.Revit.FilterRules.FilterRule)ParameterValuePresenceRuleFromRevit(revitViewFilter, revitRule); }
+                // FILTER INVERSE RULE
```

Comment:
@pawelbaran,
the HasValueFilterRule and HasNoValueFilterRule are managedby exclusion using the code line bhomParamValuePresenceRule.IsPresent = (revitRule.GetType() == typeof(HasValueFilterRule)) ? true : false;

In the Revit API HasValueFilterRule and HasNoValueFilterRule inherit both from ParameterValuePresenceRule. 
![image](https://github.com/user-attachments/assets/d185c827-5634-44d6-bdf5-b8418e45ee54)

The two Revit API classes above are managed by our BHoM class ParameterValuePresenceRule via the boolean attribute isPresent.
![image](https://github.com/user-attachments/assets/85cd6498-c15a-4bc6-8090-a7da43aac0bf)

I would say that it's doing what we are expecting but let me know in case I'm missing something ;)

Reply:
What I think you are missing here is the following line:
```C#
else if (revitRule.GetType() == typeof(Autodesk.Revit.DB.ParameterValuePresenceRule))
```

It checks whether the rule type is exactly `ParameterValuePresenceRule`, ignoring inheritance, so it will only catch this exact type. Meanwhile, its children (`HasNoValueFilterRule`, `HasValueFilterRule`) will not be caught. If you want to capture all 3 types at once, then instead of `GetType() ==` you should rather use `is` statement.